src/App.jsx

```diff
@@ -14,45 +16,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_ALPHABETICALLY = 'alphabet';
+const SORT_BY_LENGTH = 'good length';
+
+function getPreparedGoods(goods, { sortCase, isReversed }) {
+  const preparedGoods = [...goods];
+
+  if (sortCase) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortCase) {
+        case SORT_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversed ? preparedGoods.reverse() : preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortCase, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const sortedGoods
+    = getPreparedGoods(goodsFromServer, { sortCase, isReversed });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => (setSortField(SORT_ALPHABETICALLY))}
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortCase !== SORT_ALPHABETICALLY,
+          })}
```

Comment:
```suggestion
          className={cn(
            'button',
            'is-info',
            { 'is-light': sortCase !== SORT_ALPHABETICALLY }
          )}
```
Let's improve readability, fix in all places